src/main/java/fr/jcgay/maven/plugin/buildplan/AbstractLifecycleMojo.java

```diff
@@ -42,4 +60,24 @@ protected MavenExecutionPlan calculateExecutionPlan() throws MojoFailureExceptio
             throw new MojoFailureException(String.format("Cannot calculate Maven execution plan, caused by: %s", e.getMessage()), e);
         }
     }
+    
+    protected void handleOutput(final String output) {
+        if (outputFile == null) {
+            getLog().info(output);
+        } else {
+            synchronized(outputFile) {
```

Comment:
FYI synchronizing here is not enough, the `outputFile` reference will be different in each maven modules.